compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -1265,24 +1266,89 @@ public void visit(BLangAssignment assignNode) {
     }
 
     public void visit(BLangRecordDestructure stmt) {
+        List<BLangExpression> varRefs = stmt.varRef.recordRefFields.stream()
```

Comment:
The same logic is repeated in later. I think we can simplify this.

Reply:
fixed in 1bf5446